### ISSUE TITLE
Middle Clicking stops working but continues to run

### ISSUE THREAD

**Issue by rxhanson** [rxhanson/Middle-Community#45]
This is not trivial to troubleshoot, since there's not really any simple way for me to reproduce it.

So, when this happens, I'll take as much information as I can get.

1. What gestures (Magic Mouse & trackpad) are configured?
2. Is there anything that might appear to trigger this happening?
3. Are you running any other apps that listen to touch data?

If you experience this with the Magic Trackpad, you can switch the app to use Apple's private multitouch framework with this terminal command and an app restart:

`defaults write com.knollsoft.Middle usePrivateFramework -bool true`

**Comment by jpaulickcz** [comment]
I usually only notice this after I wake up the laptop (close/open lid).

2017 MacBook Pro (Intel), macOS 12.2.1

1) In what sense/where? 
2) I have not yet noticed any pattern except it's usually down after wake up, but will pay close attention now.
3) Nope.

**Comment by yehhor** [comment]
Hi! I am experiencing an issue like this with middle. 
I am using mbp 2021, after awaking laptop middle starts (i see the icon working) but middleclick does not work. After enabling Viewer - it shows clicked 1 finger tap and has not response. clicking relaunch makes everything fine.
Any ideas?

**Comment by rxhanson** [comment]
@yehhor Did you try enabling the setting for "Delay relaunch after wake"?

In the About window, select Debug -> Delay Relaunch After Wake

For some users, if the app relaunches too soon then macOS does not properly send all the touch events to the app, and there is no way for the app to determine this.

**Comment by yehhor** [comment]
> @yehhor Did you try enabling the setting for "Delay relaunch after wake"?
> 
> In the About window, select Debug -> Delay Relaunch After Wake
> 
> For some users, if the app relaunches too soon then macOS does not properly send all the touch events to the app, and there is no way for the app to determine this.

yes, this option is enabled. Also I deselected and selected it again as you mentioned before.
I see that icon appears after ~5 secs after laptop awake.

**Comment by rxhanson** [comment]
@yehhor thanks for the update. 

Do you notice that the middle click is not registered in certain apps, or it's not registered anywhere at all until a restart?

As a sanity check, you can use this website to see if the middle click is registered there or not: https://unixpapa.com/js/testmouse.html

It sounds like touch events are being sent to the app (although there might be something wrong with the events that are being sent). 

One more thing we can try up front is to increase that time period before the app will restart itself. For that, you'll need to execute a terminal command:
defaults write com.knollsoft.Middle restartOnWakeDelay -float 10

I have heard of some users that needed > 5, but have not heard of anyone that needed > 10, so setting it to 10 should be a good start point.

**Comment by yehhor** [comment]
here is the log of testmouse app:

```
mousedown   which=1 button=0 buttons=1
mousedown   which=2 button=1 buttons=1
mouseup     which=2 button=1 buttons=1
mouseup     which=1 button=0 buttons=0
```

I`ve entered that command, and it worked fine, I`ll update it if I have some issues.
